scripts/check_go_version.sh

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+fix=0
+args="$(getopt -o "" -l fix -- "$@")"
+eval set -- "$args"
+while true; do
+	case "$1" in
+	--fix)
+		fix=1
+        shift
+		;;
+	--)
+		shift
+		break
+		;;
+	*)
+		error "Unrecognized option: $1"
+		;;
+	esac
+done
+
+MOD_VERSION=$(go mod edit -json | jq -r .Go)
+echo "go.mod version: $MOD_VERSION"
+STATUS=0
+
+if [[ $fix -eq 1 ]]; then
+    for wf in .github/workflows/*.{yml,yaml}; do
+        sed -i "s/go-version:.*/go-version: \"${MOD_VERSION}\"/g" "${wf}"
+    done
+    exit $STATUS
+fi
+
+for wf in .github/workflows/*.{yml,yaml}; do
+    WF_VERSIONS=$(yq -r '.jobs[].steps[] | select(.with["go-version"]) | .with["go-version"]' -o=tsv "$wf" | grep -v '^---$' || true)
+    if [[ -z "$WF_VERSIONS" ]]; then
+        continue
+    fi
+
+    UNIQUE_WF_VERSIONS=$(echo "$WF_VERSIONS" | sort -u)
```

Comment:
```suggestion
    UNIQUE_WF_VERSIONS=$(sort -u <<<"$WF_VERSIONS")
```
Pet peeve: No need for pipe.

Reply:
1547929